### PR TITLE
refactor: remove the remaining commons library usages of InternalFuture

### PR DESCRIPTION
**Pull request by jkwatson-verta**
## Impact and Context
This will break the registry, but I have a branch ready to go to fix it.

## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

**Review by gustavoteixeirah — APPROVED** [review]

File: backend/server/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java [hunk 1]
```diff
@@ -205,7 +205,9 @@ public InternalFuture<List<KeyValue>> getAttributes(GetAttributes request) {
             unused ->
                 checkProjectPermission(projectId, ModelDBActionEnum.ModelDBServiceActions.READ),
             executor)
-        .thenCompose(unused -> attributeHandler.getKeyValues(projectId, keys, getAll), executor);
+        .thenCompose(
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:208]: Could this be `thenSupply()`?

File: backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 2]
```diff
@@ -359,7 +366,9 @@ public InternalFuture<List<KeyValue>> getAttributes(GetAttributes request) {
 
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
-        .thenCompose(unused -> attributeHandler.getKeyValues(runId, keys, getAll), executor);
+        .thenCompose(
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:369]: Could this also be `thenSupply`?

File: backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 3]
```diff
@@ -348,7 +352,10 @@ public InternalFuture<List<KeyValue>> getHyperparameters(GetHyperparameters requ
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
         .thenCompose(
-            unused -> hyperparametersHandler.getKeyValues(runId, Collections.emptyList(), true),
+            unused ->
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:355]: Same question

File: backend/server/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java [hunk 4]
```diff
@@ -680,7 +681,9 @@ public InternalFuture<GetAttributes.Response> getExperimentAttributes(GetAttribu
                 futureProjectDAO.checkProjectPermission(
                     projectIdFromExperimentMap.get(expId), ModelDBServiceActions.READ),
             executor)
-        .thenCompose(unused -> attributeHandler.getKeyValues(expId, keys, getAll), executor)
+        .thenCompose(
+            unused -> attributeHandler.getKeyValues(expId, keys, getAll).toInternalFuture(),
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java:685]: Same

File: backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 5]
```diff
@@ -339,7 +339,11 @@ public InternalFuture<List<KeyValue>> getMetrics(GetMetrics request) {
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
         .thenCompose(
-            unused -> metricsHandler.getKeyValues(runId, Collections.emptyList(), true), executor);
+            unused ->
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:342]: Same

**Review by eheinlein-verta — APPROVED** [review]
My comments are just nitpicks that I added since I saw you changes some of the usages of `thenCompose(unused ->...` but not others.

**Review by marcelo-verta-ai — APPROVED** [review]

File: backend/server/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java (hunk 1 above)
Comment (jkwatson-verta) [reply backend/server/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:208]: definitely. I stopped trying to fix everything I saw after a while. :)

**Pull request merged**